### DISCUSSION TITLE
long loads in 2sf playback

### DISCUSSION THREAD

**Discussion by Sivecano** [audacious-media-player/audacious#83] [General]
when playing 2sf files using the 2sf plugin it may take up to 10 seconds to switch to the next song.
in the meantime the previous will start playing from the start again. 
I find this rather jarring.
is this a hard limitation? if no how/where would I contribute to try and fix this issue?

**Comment by jlindgren90** [comment]
This sounds like a bug in the plugin.

**Comment by ghost** [comment]
yea i have this too, the delay gets longer the more files you play

**Comment by KynikossDragonn** [comment]
I've also been experiencing this; if I switch songs I momentarily hear previous frames leading into the song I want to actually play.

Also, the old mini2sf set for "Advance Wars: Days of Ruin" doesn't work anymore, it just uses up 100% CPU and no audio is output.

It used to work fine several years ago, I can't tell what changed with the "xSF" plugin that caused this to break.

The plugin name is also a massive misnomer, "xSF" would imply it's supporting a lot more than just the antiquated 2SF rip format using vio2sf code; a name like this would imply being able to playback _literally everything_ around the whole "PSF" concept: "PSF/PSF2/GSF/QSF/DSF/SSF/2SF/NCSF" etc...

**Reply by ahigerd** [reply]
For the record it isn't using vio2sf anymore. There might be some distant relationship with the old vio2sf codebase but I started over with a fresh fork of Desmume and CyberBotX's xSF file format parser.

I'm aware that compatibility with some REALLY old 2sf sets isn't 100%. They're bad rips and you should find a new one -- as far as I know, every rip that would have been affected by the compatibility break has been re-ripped since then.

I agree completely on the name of the plugin being misleading. That was something I'm not sure I should change, because I don't know what it might imply for people updating the plugin. (I'm not going to rule out the idea of adding support for the other formats, but I tried updating SSF/DSF and discovered that the current implementations are really fragile.)

**Comment by jlindgren90** [comment]
Pull requests, to fix the issues you've described, are welcome.

**Comment by radioactiveman** [comment]
Hi @ahigerd, would you be willing to look into these issues with the XSF plugin? 

I'm asking you because you have updated a lot of the plugin code in https://github.com/audacious-media-player/audacious-plugins/pull/97 a few years ago and seem to be familiar with it and 2SF files in general.

P.S. I can reproduce the slow title changes with files downloaded from https://www.zophar.net/music/nds.html.

Any help is welcome, thanks. :+1:

**Reply by ahigerd** [reply]
Yeah, I'm the upstream 2sf maintainer. I can take a look.
